src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/CoffObjectWriter.cs

```diff
@@ -0,0 +1,1126 @@
+
+                stream.Write(buffer);
+            }
+        }
+
+        private enum CoffComdatSelect
+        {
+            IMAGE_COMDAT_SELECT_NODUPLICATES = 1,
+            IMAGE_COMDAT_SELECT_ANY = 2,
+            IMAGE_COMDAT_SELECT_SAME_SIZE = 3,
+            IMAGE_COMDAT_SELECT_EXACT_MATCH = 4,
+            IMAGE_COMDAT_SELECT_ASSOCIATIVE = 5,
+            IMAGE_COMDAT_SELECT_LARGEST = 6,
+        }
+
+        private sealed class CoffSectionSymbol : CoffSymbolRecord
+        {
+            public uint SizeOfRawData { get; set; }
+            public ushort NumberOfRelocations { get; set; }
+            public ushort NumberOfLineNumbers { get; set; }
+            public uint CheckSum { get; set; }
+            public uint Number { get; set; }
+            public CoffComdatSelect Selection { get; set; }
+
+            private const int RegularSize =
+                sizeof(uint) +   // SizeOfRawData
+                sizeof(ushort) + // NumberOfRelocations
+                sizeof(ushort) + // NumberOfLineNumbers
+                sizeof(uint) +   // CheckSum
+                sizeof(ushort) + // Number
+                sizeof(byte) +   // Selection
+                3;               // Reserved
+
+            private const int BigObjSize = RegularSize + 2;
+
+            public override void Write(Stream stream, CoffStringTable stringTable, bool isBigObj)
+            {
+                Span<byte> buffer = stackalloc byte[isBigObj ? BigObjSize : RegularSize];
+
+                buffer.Clear();
+                BinaryPrimitives.WriteUInt32LittleEndian(buffer, SizeOfRawData);
+                BinaryPrimitives.WriteUInt16LittleEndian(buffer.Slice(4), NumberOfRelocations);
+                BinaryPrimitives.WriteUInt16LittleEndian(buffer.Slice(6), NumberOfLineNumbers);
+                BinaryPrimitives.WriteUInt32LittleEndian(buffer.Slice(8), CheckSum);
+                BinaryPrimitives.WriteUInt16LittleEndian(buffer.Slice(12), (ushort)Number);
+                buffer[14] = (byte)Selection;
+                if (isBigObj)
+                {
+                    BinaryPrimitives.WriteUInt16LittleEndian(buffer.Slice(16), (ushort)(Number >> 16));
+                }
+
+                stream.Write(buffer);
+            }
+        }
+
+        private sealed class CoffStringTable : StringTableBuilder
+        {
+            public new uint Size => (uint)(base.Size + 4);
+
+            public new uint GetStringOffset(string text)
+            {
+                return base.GetStringOffset(text) + 4;
+            }
+
+            public new void Write(FileStream stream)
+            {
+                Span<byte> stringTableSize = stackalloc byte[4];
+                BinaryPrimitives.WriteUInt32LittleEndian(stringTableSize, Size);
+                stream.Write(stringTableSize);
+                base.Write(stream);
+            }
+        }
+
+        private static class JamCrc32
+        {
+            private static uint[] Table =
```

Comment:
nit: lookup table can be replace with:

```c#
using System.Numerics;
...
private static readonly uint[] s_crcLookup = Crc32ReflectedTable.Generate(0xEDB88320u);
```

```csproj
<Compile Include="$(LibrariesProjectRoot)Common\src\System\Numerics\Crc32ReflectedTable.cs" />
```

Reply:
Good idea, I will test it!

Reply:
The polynomial is different, `0x04C11DB7` according to my notes, and there may be some other small differences. I'll try to make a test project to replicate the table with `Crc32ReflectedTable.Generate`.

Reply:
I tested it on fiddle: https://dotnetfiddle.net/SJW4qF :)

Reply:
aside:
```c#
System.IO.Hashing.Crc32 c = new();
c.Append(stream);
var result = ~BitConverter.ToUInt32(c.GetCurrentHash());
```
result matches JAMCRC from https://crccalc.com but `class JamCrc32` results are a bit off. If it's a bug, we can replace the whole class?

Reply:
`System.IO.Hashing.Crc32` uses different intial state (~0) than the COFF CRC (0). That's why it cannot be used directly.